### ISSUE TITLE
error message "logger:remove_handler(default) failed" printed at startup

### ISSUE THREAD

**Issue by slang800** [erlang-lager/lager#507]
I found a small issue where upgrading [amqp_client](https://hex.pm/packages/amqp_client) from `3.7.13` to `3.7.14` causes an error message to be printed when I start up my Elixir application:

```
[error] calling logger:remove_handler(default) failed: :error {:badmatch, {:error, {:not_found, :default}}}
```

The error is being printed due to code introduced by [this change](https://github.com/erlang-lager/lager/commit/9f8d7a705ef5e2e263f1b1b549ec8ba4d3f59a36), since lager is a dependency of amqp_client and was upgraded along with amqp_client. However, I have no idea if this is a defect in lager or if it's another library using lager incorrectly and I'm not sure how to check.

Is this the right place to report this?

**Comment by hauleth** [comment]
@notslang yes, this is due to fact that Elixir is also removing `default` handler if you have `handle_otp_reports` set to `true` (the default). So you have 2 options:

- live with it
- set `handle_otp_reports` to `false`

**Comment by jadencodes** [comment]
This is a bit late, but I ran into the same thing; there is another potential fix for this mentioned at the bottom of the [amqp client](https://github.com/pma/amqp) page. You need to add `:lager` to start before `:logger`. So your file will look something like this:

```elixir
def application do
    [
      extra_applications: [:lager, :logger, :amqp]
    ]
  end
```

Hope this is useful.
